pkg/datasources/procedures.go

```diff
@@ -84,10 +84,18 @@ func ReadContextProcedures(ctx context.Context, d *schema.ResourceData, meta int
 	}
 	procedures, err := client.Procedures.Show(ctx, req)
 	if err != nil {
-		return diag.FromErr(err)
-	}
+		id := d.Id()
```

Comment:
When Show fails in the first run, then the error will print empty parens, because `d.Id()` would be empty. Maybe instead of id, use values from schema `databaseName/schemaName`. Additionally, error message says `in schema`, but it can also be `in database` depending on the configuration

Reply:
updated to `fmt.Sprintf(`%v|%v`, databaseName, schemaName)`. This data source needs databaseName and schemaName not to be required to do as you are describing. I think that would be a great improvement, but also we need to do this for all other data sources as well and we should discuss what the schema should look like. I don't think this PR is the place to do this.